src/shader_program.cpp

```diff
@@ -0,0 +1,88 @@
+#include "shader_program.hpp"
+#include "shader.hpp"
+
+void ShaderProgram::checkErrors_() const noexcept {
+  GLint success;
+  glGetProgramiv(id_, GL_LINK_STATUS, &success);
+  if (!success) {
+    GLchar info_log[512];
+    glGetProgramInfoLog(id_, sizeof(info_log), nullptr, info_log);
+    ERROR_LOG("failed linking shader program.");
+  }
+}
+
+ShaderProgram::ShaderProgram(const char *vertex_path,
+                             const char *fragment_path) {
+  VertexShader vertex_shader(vertex_path);
+  FragmentShader fragment_shader(fragment_path);
+
+  id_ = glCreateProgram();
+  glAttachShader(id_, vertex_shader.getId());
+  glAttachShader(id_, fragment_shader.getId());
+  glLinkProgram(id_);
+
+  checkErrors_();
```

Comment:
Thank you for the comment, since I declared and implemented a destructor for shaders(line 58 in shader.cpp), once they get out of scope the corresponding OpenGL function will delete them.